src/common/components/build-history/index.js

```diff
@@ -4,21 +4,26 @@ import { connect } from 'react-redux';
 import BuildRow from '../build-row';
 import { Message } from '../forms';
 
-const BuildHistory = (props) => {
-  const { repository, success } = props;
-
-  const builds = props.builds.map((build) => (
-    <BuildRow key={build.buildId} {...build} repository={repository} />
-  ));
-
-  return success ? (
-    <div>
-      {builds.length > 0
-        ? builds
-        : <Message status='info'>This snap has not been built yet.</Message>
-      }
-    </div>
-  ) : null;
+export const BuildHistory = (props) => {
+  const { repository, success, builds } = props;
+
+  const hasBuilds = (builds && builds.length > 0);
+
+  if (!success) {
+    return null;
+  }
+
+  if (!hasBuilds) {
+    return <Message status='info'>This snap has not been built yet.</Message>;
+  }
+
+  const buildRows = builds
+    .sort((a,b) => (+a.buildId < +b.buildId))
```

Comment:
This will produce wrong results depending on what pairs of elements the comparison function happens to be called with (because if `a.buildId` is greater than `b.buildId`, then this will incorrectly return 0 rather than something negative).  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort suggests using subtraction as the comparison function for integers, so maybe `(+b.buildId) - (+a.BuildId)` instead?